psycopg/psycopg/connection_async.py

```diff
@@ -111,27 +111,46 @@ async def connect(
         timeout = timeout_from_conninfo(params)
         rv = None
         attempts = await conninfo_attempts_async(params)
+        connection_errors: list[tuple[e._NoTracebackException, str]] = []
         for attempt in attempts:
             try:
                 conninfo = make_conninfo("", **attempt)
                 gen = cls._connect_gen(conninfo, timeout=timeout)
                 rv = await waiting.wait_conn_async(gen, interval=_WAIT_INTERVAL)
             except e._NO_TRACEBACK as ex:
-                if len(attempts) > 1:
-                    logger.debug(
-                        "connection attempt failed: host: %r port: %r, hostaddr %r: %s",
-                        attempt.get("host"),
-                        attempt.get("port"),
-                        attempt.get("hostaddr"),
-                        str(ex),
-                    )
-                last_ex = ex
+                attempt_details = "host: {}, port: {}, hostaddr: {}".format(
+                    repr(attempt.get("host")),
+                    repr(attempt.get("port")),
+                    repr(attempt.get("hostaddr")),
+                )
+                connection_errors.append((ex, attempt_details))
             else:
                 break
 
         if not rv:
-            assert last_ex
-            raise last_ex.with_traceback(None)
+            num_errors = len(connection_errors)
+            assert num_errors > 0
+
+            last_exception, _ = connection_errors[-1]
+            if num_errors == 1:
+                raise last_exception.with_traceback(None)
+            else:
+                formatted_attempts = []
+                for error, attempt_details in connection_errors:
+                    formatted_attempts.append(f"- {attempt_details}: {str(error)}")
```

Comment:
`str()` is implicit
```suggestion
                    formatted_attempts.append(f"- {attempt_details}: {error}")
```

Reply:
Addressed in [982dda4](https://github.com/psycopg/psycopg/commit/982dda4f5c99832d95e0e447ba6c16eb9122a24a)